### ISSUE TITLE
About Training Loss

### ISSUE THREAD

**Issue by Ferry1231** [LTH14/mar#37]
I am using my self-trained autoencoder as the encoder to train on the CIFAR-10 dataset. After 500 epochs, the loss dropped to around 0.1, but the reconstructed images are almost all white, with pixel values being quite high. I observed that the sample_tokens before decoding in the AE after sampling had very large values, with a mean reaching over 1000, while the original mean in the latent space was only about 2. I’m not sure why this is happening, and I would greatly appreciate your help in resolving this issue.

**Comment by LTH14** [comment]
500 epochs on CIFAR-10 might not be enough. Since the ema is 0.9999, the model with ema needs around 100k iterations to generate a reasonable image. The model without ema still needs around 50k iterations. Also, please check 1. whether you have normalized the tokens according to your new autoencoder (the current normalization is 0.2325, which is specific to our ImageNet tokenizer) 2. you can use 1000 diffusion steps instead 100 to see whether the large value is because of the diffusion process

**Comment by Ferry1231** [comment]
It means i need to train MAR on cifar10 at least for 100k epochs?? 😱

**Comment by LTH14** [comment]
No -- 100k iterations (160 epochs on ImageNet with bsz=2048). For CIFAR10 it should be around 2000 epochs.

**Comment by Ferry1231** [comment]
ok thank you very much

**Comment by 1359347500cwc** [comment]
Have you managed to resolve this issue? I am also using my self-trained autoencoder as both the encoder and decoder to train on the ImageNet dataset. After 270 epochs, the loss has decreased to around 0.3, but the generated images are almost entirely white.


here is the eval.sh

torchrun \
    --node_rank=$RANK \
    --nproc_per_node=8 \
    --nnodes=1 \
    --master_addr="${MASTER_ADDR}" \
    --master_port=12344 \
    main_mar.py \
    --model mar_base --diffloss_d 6 --diffloss_w 1024 \
    --eval_bsz 32 --num_images 50000 \
    --num_iter 256 --num_sampling_steps 100 --cfg 2.9 --cfg_schedule linear --temperature 1.0 \
    --output_dir /mar_result/mar_mamba_base_0208 \
    --resume base_0208/checkpoint-270.pth \
    --data_path ${IMAGENET_PATH} --evaluate

**Comment by jihopark00** [comment]
I'm observing same issue here, using self-trained Auto Encoder, which normalizes the feature (mean value is about 0.6). Training is stabilized fastly but the sampled token's abs().mean() values is around 500, which is too large.

Wondering if there's required conditions on target distirbution for current diffusion formulation
